source/ide.js

```diff
@@ -1660,6 +1675,42 @@ function fileDlg(title, filename, allowNewFilename, onOkay)
 			}
 		}
 	}
+
+	function download(filename, text) {
+		var element = document.createElement('a');
+		element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
+		element.setAttribute('download', filename);
+	  
+		element.style.display = 'none';
+		document.body.appendChild(element);
+	  
+		element.click();
+	  
+		document.body.removeChild(element);
+	}
+
+	function exportFile(filename){
+		let data = localStorage.getItem("tscript.code." + filename);
+		download(filename + ".tscript", data);
+	}
+
+	function importFile(){
+		let fileImport = document.getElementById('file-import');
+		fileImport.addEventListener('change', async (event) => {
+			if(event.target.files){
+				var file = event.target.files[0];
+				var filename = file.name.split('.tscript')[0];
+				var data = await file.text();
+				localStorage.setItem("tscript.code." + filename, data);
+				files.push(filename);
+				var option = document.createElement('option');
+				option.text = filename;
+				list.add(option);
```

Comment:
[`HTMLSelectElement`](https://developer.mozilla.org/docs/Web/API/HTMLSelectElement) does't have an `add` method. You probably meant `list.options.add(option)`, but just changing this won't work either, because `list` might not be an `HTMLSelectElement` at all.

When there are no saved files, `list` is created as `<text>No documents saved.</text>` ([here](https://github.com/Tilo-K/tscript/blob/c1e0ab9692eb8d1928d56a143f5efad46d27a24c/source/ide.js#L1533)). This means that in this case, `list` is an [`HTMLUnknownElement`](https://developer.mozilla.org/docs/Web/API/HTMLUnknownElement), which obviously doesn't have an `options` parameter:

![The resulting TypeError](https://user-images.githubusercontent.com/28811733/99797849-491d1780-2b30-11eb-9b8c-e91426dd352c.png)